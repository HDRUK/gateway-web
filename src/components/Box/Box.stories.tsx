import type { Meta } from "@storybook/react";
import { BoxProps } from "@mui/material/Box";
import Box from "./Box";

/**
 * Documents:
 * https://mui.com/material-ui/react-box/
 */

const meta: Meta<typeof Box> = {
    component: Box,
    title: "Layout/Box",
};

export default meta;

export const Playground = (args: BoxProps) => <Box {...args} />;

Playground.args = {
    children: "This is a box",
};
