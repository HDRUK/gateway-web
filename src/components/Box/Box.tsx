import MuiBox, { BoxProps as MuiBoxProps } from "@mui/material/Box";

const Box = (props: MuiBoxProps) => {
    const { sx, ...rest } = props;
    return (
        <MuiBox
            sx={{
                bgcolor: "white",
                p: 2,
                ...sx,
            }}
            {...rest}
        />
    );
};

export default Box;
