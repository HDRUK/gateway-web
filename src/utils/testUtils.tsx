import React, { ReactNode } from "react";
import {
    render,
    renderHook,
    RenderHookOptions,
    RenderOptions,
    RenderResult,
    RenderHookResult,
} from "@testing-library/react";
import { SWRConfig } from "swr";
import { CacheProvider, ThemeProvider } from "@emotion/react";
import ActionBarProvider from "@/providers/ActionBarProvider";
import DialogProvider from "@/providers/DialogProvider";
import theme from "@/config/theme";
import createEmotionCache from "@/config/createEmotionCache";

const clientSideEmotionCache = createEmotionCache();

const Wrapper = ({ children }: { children: ReactNode }) => {
    return (
        <SWRConfig
            value={{
                provider: () => new Map(),
            }}>
            <CacheProvider value={clientSideEmotionCache}>
                <ThemeProvider theme={theme}>
                    <ActionBarProvider>
                        <DialogProvider>{children}</DialogProvider>
                    </ActionBarProvider>
                </ThemeProvider>
            </CacheProvider>
        </SWRConfig>
    );
};

interface OptionProps
    extends RenderOptions<
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement,
        HTMLElement
    > {
    wrapperProps: Record<string, unknown>;
}

const customRender = (
    ui: React.ReactElement<
        unknown,
        string | React.JSXElementConstructor<unknown>
    >,
    options?: OptionProps
): RenderResult => {
    const { wrapperProps, ...rest } = options || {};

    return render(ui, {
        wrapper: props => <Wrapper {...props} {...wrapperProps} />,
        ...rest,
    });
};

interface OptionHookProps
    extends RenderHookOptions<
        unknown,
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement,
        HTMLElement
    > {
    wrapperProps: Record<string, unknown>;
}

const customRenderHook = (
    ui: (initialProps: unknown) => unknown,
    options?: OptionHookProps
): RenderHookResult<unknown, unknown> => {
    const { wrapperProps, ...rest } = options || {};
    return renderHook(ui, {
        wrapper: props => <Wrapper {...props} {...wrapperProps} />,
        ...rest,
    });
};

export * from "@testing-library/react";

export { customRenderHook as renderHook };
export { customRender as render };
