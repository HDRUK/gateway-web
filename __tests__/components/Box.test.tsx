import React from "react";
import Box from "@/components/Box";
import { render } from "../testUtils";

describe("Box", () => {
    it("should render component", async () => {
        const wrapper = render(<Box>Box content</Box>);

        expect(wrapper.container).toMatchSnapshot();
    });
});
