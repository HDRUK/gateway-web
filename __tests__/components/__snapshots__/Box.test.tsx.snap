// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should render component 1`] = `
<div>
  <div
    class="MuiBox-root mui-style-1ytu6zl"
  >
    Box content
  </div>
</div>
`;
